src/vs/workbench/api/node/extHostQuickOpen.ts

```diff
@@ -117,4 +122,50 @@ export class ExtHostQuickOpen implements ExtHostQuickOpenShape {
 		}
 		return undefined;
 	}
+
+	// ---- workspace folder picker
+
+	showWorkspaceFolderPick(folders: WorkspaceFolder[], options?: WorkspaceFolderPickOptions, token = CancellationToken.None): Thenable<WorkspaceFolder> {
```

Comment:
👎 this just duplicates way to much code of `showQuickPick` for not having a private method that does the work and that both `showQuickPick` and `showWorkspaceFolderPick` call.

Reply:
@jrieken can you make clear what you would want to share between the 2 methods? since the one does a complicated dance of converting the passed in items to pick items,  I did not see much value and extracting a method there.